pkg/assembler/backends/testing/isVulnerability.go

```diff
@@ -0,0 +1,171 @@
+
+	for _, vuln := range c.isVulnerability {
+		if vuln.Justification == justification && vuln.Osv == selectedOsv {
+			if val, ok := vuln.Vulnerability.(model.Cve); ok {
+				if &val == selectedCve {
+					return
+				}
+			} else if val, ok := vuln.Vulnerability.(model.Ghsa); ok {
+				if &val == selectedGhsa {
+					return
+				}
+			}
+		}
+	}
+
+	newIsVuln := &model.IsVulnerability{
+		Osv:           selectedOsv,
+		Justification: justification,
+		Origin:        "testing backend",
+		Collector:     "testing backend",
+	}
+	if selectedCve != nil {
+		newIsVuln.Vulnerability = selectedCve
+	} else {
+		newIsVuln.Vulnerability = selectedGhsa
+	}
+
+	c.isVulnerability = append(c.isVulnerability, newIsVuln)
+}
+
+// Query CertifyPkg
+
+func (c *demoClient) IsVulnerability(ctx context.Context, isVulnerabilitySpec *model.IsVulnerabilitySpec) ([]*model.IsVulnerability, error) {
+
+	if isVulnerabilitySpec.Cve != nil && isVulnerabilitySpec.Ghsa != nil {
+		return nil, gqlerror.Errorf("cannot specify cve and ghsa together for IsVulnerability")
+	}
+
+	var foundIsVulnerability []*model.IsVulnerability
+
+	for _, h := range c.isVulnerability {
+		cveMatchOrSkip := false
+		osvMatchOrSkip := false
+		ghsaMatchOrSkip := false
+		justificationMatchOrSkip := false
+		collectorMatchOrSkip := false
+		originMatchOrSkip := false
+
+		if isVulnerabilitySpec.Justification == nil || h.Justification == *isVulnerabilitySpec.Justification {
+			justificationMatchOrSkip = true
+		}
+
+		if isVulnerabilitySpec.Collector == nil || h.Collector == *isVulnerabilitySpec.Collector {
+			collectorMatchOrSkip = true
+		}
+		if isVulnerabilitySpec.Origin == nil || h.Origin == *isVulnerabilitySpec.Origin {
+			originMatchOrSkip = true
+		}
+
+		if isVulnerabilitySpec.Osv == nil {
+			osvMatchOrSkip = true
+		} else if isVulnerabilitySpec.Osv != nil && h.Osv != nil {
+			newOSV, err := filterOSVID(h.Osv, isVulnerabilitySpec.Osv)
+			if err != nil {
+				return nil, err
+			}
+			if newOSV != nil {
+				osvMatchOrSkip = true
+			}
+		}
+
+		if isVulnerabilitySpec.Cve == nil {
+			cveMatchOrSkip = true
+		} else if isVulnerabilitySpec.Cve != nil {
+			if val, ok := h.Vulnerability.(*model.Cve); ok {
+				if isVulnerabilitySpec.Cve.Year == nil || val.Year == *isVulnerabilitySpec.Cve.Year {
+					newCve, err := filterCVEID(val, isVulnerabilitySpec.Cve)
+					if err != nil {
+						return nil, err
+					}
+					if newCve != nil {
+						cveMatchOrSkip = true
+					}
+				}
+			}
+		}
+
+		if isVulnerabilitySpec.Ghsa == nil {
+			ghsaMatchOrSkip = true
+		} else if isVulnerabilitySpec.Ghsa != nil {
+			if val, ok := h.Vulnerability.(*model.Ghsa); ok {
+				newGhsa, err := filterGHSAID(val, isVulnerabilitySpec.Ghsa)
+				if err != nil {
+					return nil, err
+				}
+				if newGhsa != nil {
+					ghsaMatchOrSkip = true
+				}
+			}
+		}
+
+		if justificationMatchOrSkip && collectorMatchOrSkip && originMatchOrSkip && osvMatchOrSkip && cveMatchOrSkip && ghsaMatchOrSkip {
```

Comment:
nit: probably maybe use a bool and then OR the bool throughout instead unless its being used for other things as well

Reply:
Not sure I understand?

Reply:
i mean instead of having individual bools, we have one bool

matched := true

and then each check it goes through it does
match &= false if it does not pass the check

Reply:
Ah, yea sure that can be done. This was done to address a comment in a previous PR to see what checks are being done.

Reply:
Will address in final PR